ui/app/hooks/useTimeout.js

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, useRef, useCallback } from 'react'
+
+/**
+ * useTimeout
+ *
+ * @param {Function}  cb   - callback function inside setTimeout
+ * @param {number}  delay   - delay in ms
+ * @param {boolean}  immediate   - delay in ms
+ *
+ * @return {Function}
+ */
+export function useTimeout (cb, delay, immediate = true) {
+  const saveCb = useRef()
+  const [triggered, setTriggered] = useState(immediate)
+
+  useEffect(() => {
+    saveCb.current = cb
+  }, [cb])
+
+  useEffect(() => {
+    if (!triggered) {
+      return
+    }
+
+    const id = setTimeout(() => {
+      setTriggered(false)
+      saveCb.current()
+    }, delay)
+
+    return () => {
+      clearTimeout(id)
+    }
+  }, [delay, triggered])
+
+  const trigger = useCallback(() => {
+    setTriggered(true)
```

Comment:
Note that this wouldn't reset the timer if the user copied the address multiple times. This could lead to strange behavior, e.g. the text rapidly flashing between states, rather than showing for 3 seconds.

Perhaps you could store the timeout id in state instead of this `triggered` boolean? Then the old timeout could be cleared here (if it's set), allowing the timer to be reset.

Reply:
thank you for the suggestions, now using `timeoutId` in a state, so that it is cleared/reset whenever `startTimeout` is called